packages/rum-core/src/domain/resource/matchRequestTiming.spec.ts

```diff
@@ -22,63 +23,75 @@ describe('matchRequestTiming', () => {
 
-    expect(timing).toEqual(undefined)
+    expect(matchingTiming).toEqual(undefined)
   })
 
   it('should not match two not following timings nested in the request ', () => {
-    const match1 = createResourceEntry({ startTime: 150 as RelativeTime, duration: 100 as Duration })
-    const match2 = createResourceEntry({ startTime: 200 as RelativeTime, duration: 100 as Duration })
-    entries.push(match1, match2)
+    const entry1 = createResourceEntry({ startTime: 150 as RelativeTime, duration: 100 as Duration })
+    const entry2 = createResourceEntry({ startTime: 200 as RelativeTime, duration: 100 as Duration })
+    entries.push(entry1, entry2)
 
-    const timing = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
+    const matchingTiming = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
 
-    expect(timing).toEqual(undefined)
+    expect(matchingTiming).toEqual(undefined)
   })
 
   it('should not match multiple timings nested in the request', () => {
-    const match1 = createResourceEntry({ startTime: 100 as RelativeTime, duration: 50 as Duration })
-    const match2 = createResourceEntry({ startTime: 150 as RelativeTime, duration: 50 as Duration })
-    const match3 = createResourceEntry({ startTime: 200 as RelativeTime, duration: 50 as Duration })
-    entries.push(match1, match2, match3)
+    const entry1 = createResourceEntry({ startTime: 100 as RelativeTime, duration: 50 as Duration })
+    const entry2 = createResourceEntry({ startTime: 150 as RelativeTime, duration: 50 as Duration })
+    const entry3 = createResourceEntry({ startTime: 200 as RelativeTime, duration: 50 as Duration })
+    entries.push(entry1, entry2, entry3)
 
-    const timing = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
+    const matchingTiming = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
 
-    expect(timing).toEqual(undefined)
+    expect(matchingTiming).toEqual(undefined)
   })
 
   it('should not match invalid timing nested in the request ', () => {
-    const match = createResourceEntry({
-      duration: 100 as Duration,
+    const entry = createResourceEntry({
+      // fetchStart < startTime is invalid
       fetchStart: 0 as RelativeTime,
       startTime: 200 as RelativeTime,
     })
-    entries.push(match)
 
-    const timing = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
+    entries.push(entry)
 
-    expect(timing).toEqual(undefined)
+    const matchingTiming = matchRequestTiming(FAKE_REQUEST as RequestCompleteEvent)
+
+    expect(matchingTiming).toEqual(undefined)
   })
 })
+
+export function createResourceEntry(overrides?: Partial<RumPerformanceResourceTiming>): PerformanceResourceTiming {
+  const rumPerformanceResourceTiming: Partial<PerformanceResourceTiming> = createPerformanceEntry(
+    RumPerformanceEntryType.RESOURCE,
+    overrides
+  )
+  return {
+    ...rumPerformanceResourceTiming,
+    toJSON: () => rumPerformanceResourceTiming,
+  } as PerformanceResourceTiming
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about having this behaviour for all performance entries?

Reply:
It would mean adding `toJSON` to RumPerformanceResourceTiming type only for this specific case?

Reply:
oh indeed, we don't have it for all performance entries.
to be noted, it will be added to resource type with the v5 and it is already in long task type.